packages/discovery-react-components/src/components/DocumentPreview/DocumentPreview.tsx

```diff
@@ -107,21 +160,44 @@ const DocumentPreview: FC<Props> = ({
 
 interface PreviewDocumentProps {
   document?: QueryResult | null;
+  file?: string;
   highlight?: any;
+  currentPage: number;
+  scale: number;
+  setPdfPageCount?: (count: number) => void;
   setLoading: (loading: boolean) => void;
   setHideToolbarControls?: (disabled: boolean) => void;
   loading: boolean;
   hideToolbarControls: boolean;
 }
 
 function PreviewDocument({
+  file,
+  currentPage,
+  scale,
   document,
   loading,
+  setPdfPageCount,
   setLoading,
   hideToolbarControls,
   setHideToolbarControls,
   highlight
 }: PreviewDocumentProps): ReactElement | null {
+  // if we have PDF data, render that
+  // otherwise, render fallback document view
+  if (file) {
+    return (
+      <PdfViewer
```

Comment:
@fukudasjp I think we can replace this with your `PdfViewerWithHighlight`, correct?

Reply:
Yes, but I recommend to do it in a separated PR because it will involve highlight location stuffs, too.